### ISSUE TITLE
RTSSpec for scala.js - to share or to dupe

### ISSUE THREAD

**Issue by wongelz** [zio/zio#548]
This issue is to discuss/track the approach to refactor the existing `RTS` unit tests to cover scala.js (#22). (This depends on #515) 

The tests in `RTSSpec` can be categorised into three groups:
1. Tests that will run in both JVM & JS and therefore can be shared (mainly synchronous code)
2. Tests that can be refactored to pass in JS by returning a `Future` to specs2 (ie. asynchronous code) - we will need to provide an `unsafeToFuture` method.
3. Tests that will only ever pass and/or only make sense on JVM and cannot be shared

However - the `AbstractRTSSpec` superclass will not work in Javascript because `AroundTimeout` makes use of `Thread.sleep`. So any shared tests will have to forgo the timeout safeguards provided by `AbstractRTSSpec`.

Given the above caveats: is it worthwhile to share *any* tests or just duplicate everything?

**Comment by jdegoes** [comment]
I think we should share but there's no rush. I'd like to switch to use `testz` for testing, in which case all this will become much simpler.

**Comment by ghostdogpr** [comment]
Closing in favor of #804
